dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.edcClient.api.DefaultApi
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
```

Comment:
verifyDataIdIsExists - das "Is" ist zu viel. 
 Also besser:
 verifyDataIdExists

Reply:
fixed